### PR TITLE
Graph Loading with new Filenames/Attributes

### PR DESCRIPTION
**Pull request by christinem**
This PR changes the way graphs are grabbed and loaded, based on the new convention of files being named after their primary IDs. 

@Ian-Stewart-Binks the IDs right now are stored within the `graph-button` elements for each graph, with the key `id` - each respective ID stored in its respective button.  To note, however, the buttons are only dynamically created when you open the 'graphs' tab in the sidebar. If you need them stored in a different place, let me know, it should not be too hard, depending on how you want to use it.

### CODE REVIEW

File: public/js/graph/setup.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ var nodes = [];               // List of all nodes
 $(document).ready(function () {
     'use strict';
 
-    loadGraph('graph-csc');
+    loadGraph(1);
```
Comment (david-yz-liu) [inline public/js/graph/setup.js]: Make this a string

File: public/js/graph/sidebar/sidebar_events.js [hunk 2]
```diff
@@ -31,10 +30,11 @@ function createGraphButtons(graphs) {
     'use strict';
 
     for (var i = 0; i < graphs.length; i++) {
-        var graphTitle = graphs[i].graph_title.toLowerCase();
+        var graphTitle = graphs[i].title.toLowerCase();
         var graphButton = '<div id = "graph-' + graphTitle +'" class = "graph-button">';
         $('#graphs').append(graphButton);
         $('#graph-' + graphTitle).html(graphTitle.toUpperCase());
```
Comment (david-yz-liu) [inline public/js/graph/sidebar/sidebar_events.js]: Why not use `graphs[i].title`?

File: public/js/graph/setup.js (hunk 1 above)
Comment (christinem) [inline public/js/graph/setup.js]: Done!

**Comment by Ian-Stewart-Binks** [comment]
@christinem Hmm. We'll need access to the current graph ID whenever the use is able to post a graph to Facebook, and we don't need access to every ID. Where do you think the best place to put it would be?

File: public/js/graph/sidebar/sidebar_events.js (hunk 2 above)
Comment (christinem) [inline public/js/graph/sidebar/sidebar_events.js]: Ah, yes I can definitely do so. Done!

**Comment by christinem** [comment]
@Ian-Stewart-Binks if it's only the current graph ID you need, we can either use a cookie to denote the active graph ID or the data could be assigned to the `graph` element itself with a key `active-graph` or something of the sort.

**Comment by david-yz-liu** [comment]
Seems like a perfect use for a cookie.

**Comment by Ian-Stewart-Binks** [comment]
@christinem Since the information doesn't need to persist throughout multiple browsing sessions*, I don't know if a cookie would be the best house for it. Perhaps it would be best stored in the `.data` of the graph element. What do you think?

*I don't believe that's the correct use of the term 'browsing session', but you get the idea. Browser session: [A browser session is a continuous period of user activity in the browser, where successive events are separated by no more than 30 minutes.](https://blog.mozilla.org/metrics/2010/12/22/browsing-sessions/)

**Comment by david-yz-liu** [comment]
If the current graph persists, users can pick up where they left off. Rather than always pull the CSC graph (or graph id '1'), check the cookie and grab that graph.

**Comment by christinem** [comment]
@Ian-Stewart-Binks I actually think a cookie might be handy. I wanted to add in the feature of, if you switch tab, and go back to the graph tab, the active graph will load. Kind of like what happens when you switch the the POSt tab and whatever POSt was opened last, that's the one that opens right now.

**Comment by christinem** [comment]
Ah @david-yz-liu beat me to it. 
I will now just add that functionality to pick up where someone left off.

**Comment by Ian-Stewart-Binks** [comment]
@david-yz-liu @christinem Hadn't thought of that. I agree.

**Comment by christinem** [comment]
Done!

File: public/js/graph/sidebar/sidebar_events.js [hunk 3]
```diff
@@ -45,26 +45,29 @@ function createGraphButtons(graphs) {
...
+    setCookie('active-graph', id);
+    
     // Remove current graph
     $('#graph').empty();
 
-    getRemote(graph.toUpperCase() + '/' + graph + '_graph.svg');
+    getRemote('static/res/graphs/' + id + '.svg');
 
-    FCEPrerequisiteCourses = [csc318, csc454];
+    // Only create this if CSC graph loaded
```
Comment (Ian-Stewart-Binks) [inline public/js/graph/sidebar/sidebar_events.js]: Good catch.

File: public/js/graph/setup.js [hunk 4]
```diff
@@ -21,7 +21,8 @@ var nodes = [];               // List of all nodes
 $(document).ready(function () {
     'use strict';
 
-    loadGraph('graph-csc');
+    var active = getCookie('active-graph');
+    loadGraph(active);
```
Comment (david-yz-liu) [inline public/js/graph/setup.js]: The cookie might not exist; you need to check for this.

Comment (christinem) [inline public/js/graph/setup.js]: Ah right, done.

File: public/js/graph/sidebar/sidebar_events.js [hunk 5]
```diff
@@ -45,26 +45,29 @@ function createGraphButtons(graphs) {
...
- * @param{string} id ID name of graph button clicked
+ * @param{string} id ID of graph in database
 **/
 function loadGraph(id) {
     'use-strict';
     
-    var graph = id.substring(6, id.length);
-
+    setCookie('active-graph', id);
+
```
Comment (david-yz-liu) [inline public/js/graph/sidebar/sidebar_events.js]: Whitespace

Comment (christinem) [inline public/js/graph/sidebar/sidebar_events.js]: Fixed.

**Comment by david-yz-liu** [comment]
@christinem Okay, two more things before I merge this branch into master:

1) The FCE count bug needs to be fixed.
2) Please play around with making the sidebar look more attractive for clicking. E.g., give it a "question mark" icon, make it a bit thicker, and give it a different colour on hover.

**Pull request merged**